src/BenchmarkDotNet/Running/ConsoleTitler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.IO;
+using BenchmarkDotNet.Portability;
+
+namespace BenchmarkDotNet.Running
+{
+    /// <summary>
+    /// Updates Console.Title, subject to platform capabilities and Console availability.
+    /// Restores the original (or fallback) title upon disposal.
+    /// </summary>
+    internal class ConsoleTitler : IDisposable
+    {
+        /// <summary>
+        /// Whether this instance has any effect. This will be false if the platform doesn't support Console retitling,
+        /// or if Console output is redirected.
+        /// </summary>
+        public bool IsEnabled { get; private set; }
+
+        private string oldConsoleTitle;
+
+        public ConsoleTitler(string initialTitle)
+        {
+            // Return without enabling if Console output is redirected.
+            if (Console.IsOutputRedirected)
+            {
+                return;
+            }
```

Comment:
I'm not 100% sure if we want to disable this if output is redirected. I can imagine a scenario where the output is piped to a file, but it would still be useful to see the progress in the title. But not a big deal to me either way.